### PR TITLE
Dark Mode - Step 1 - Base theme and style configurations

### LINKED ISSUE #1572: Dark Mode: Switch to material theme and add base styles
Switch the base theme to use google's material theme for dark mode support. This is a sub-ticket for the bigger feature to implement dark mode support in #1562 

TODO:
- Update material dependency (fix errors from nullability changes)
- Create new base theme, colors, dimens, and fonts
- Add the new material theme and set it as the AppTheme

### PR DESCRIPTION
**Pull request by AmandaRiu**
Fixes #1572 by doing the following:
- Update material dependency
- Create new `themes.xml` resource files and move theme configurations to these files.
- Update theme to use a new `Woo.DayNight` theme
- Add new color palette in `colors.xml` and moved the existing color configurations to `colors_old.xml`
- Add new `roboto-light` downloadable font config
- Add font configurations and apply to theme
- Add resource files for alpha and elevation settings for each material plane

### Misc
- Swapped out `Toolbar` for `MaterialToolbar` while configuring different theme attributes and left it in since it will be changed later anyway. 
- The toolbar will change colors for dark/light mode, but should not be evaluated for completeness as there is still much more to do for this. 

### Testing
There's not really much here to test since styles haven't been applied to anything. Updating the base theme over to using `Theme.MaterialComponents.DayNight.DarkActionBar` did cause some visual changes, but they should be ignored at this point. 
- Build the app and ensure it doesn't crash.

### CODE REVIEW

File: WooCommerce/build.gradle [hunk 1]
```diff
@@ -125,7 +125,7 @@ dependencies {
     implementation "androidx.recyclerview:recyclerview:1.1.0"
     implementation "androidx.multidex:multidex:2.0.1"
     implementation "androidx.appcompat:appcompat:1.1.0"
-    implementation "com.google.android.material:material:1.0.0"
+    implementation "com.google.android.material:material:1.2.0-alpha02"
```
Comment (nbradbury) [inline WooCommerce/build.gradle]: Are we comfortable using an alpha version of this library? On the P2 we discussed moving to v1.1.0-beta02.

**Comment by nbradbury** [comment]
Noticed we still declare `AppTheme` in [attrs.xml](https://github.com/woocommerce/woocommerce-android/blob/develop/WooCommerce/src/main/res/values/attrs.xml#L20). Should that be updated?

File: WooCommerce/build.gradle (hunk 1 above)
Comment (AmandaRiu) [reply WooCommerce/build.gradle]: Good catch! `1.1.0-rc01` is now available so I've switched to that. Thanks Nick!

**Comment by nbradbury** [comment]
@AmandaRiu Changes look good, but [this comment](https://github.com/woocommerce/woocommerce-android/pull/1746#issuecomment-567557239) hasn't been addressed.

**Comment by AmandaRiu** [comment]
@nbradbury thanks for the review. Updated the styleable attribute name to match new theme naming. Ready for another round.

**Comment by AmandaRiu** [comment]
@nbradbury Actually, I'm going to tweak a few things in this PR so hold off on another review.

**Comment by AmandaRiu** [comment]
@nbradbury Fixed some weird lint errors apparently caused by the version of the material library we are using. It's a known issue, hopefully it will be fixed in the release version. We can follow the issue [here](https://github.com/material-components/material-components-android/issues/504).

**Pull request merged**